flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/tasks/OneInputStreamTask.java

```diff
@@ -98,12 +96,9 @@ public void init() throws Exception {
 	}
 
 	@Override
-	protected void run() throws Exception {
-		// cache processor reference on the stack, to make the code more JIT friendly
-		final StreamInputProcessor<IN> inputProcessor = this.inputProcessor;
-
-		while (running && inputProcessor.processInput()) {
-			// all the work happens in the "processInput" method
+	protected void performDefaultAction(ActionContext context) throws Exception {
+		if (!inputProcessor.processInput()) {
```

Comment:
This is a blocking operation, it will block until one `StreamRecord` processed. In the meantime, other letters in mailbox such as processing timer are delayed.

Reply:
I am aware of that and that is why we have also planned FLINK-12479 to make the inputs non-blocking.
For the scope of this PR it should not matter and we don't have to wait for FLINK-12479 to complete until we work on the subtasks that use the mailbox for checkpoints etc.

Reply:
Thanks, I did notice that.